ui/libproject/devstart.rb

```diff
@@ -0,0 +1,86 @@
+require "optparse"
+require_relative "utils/common"
+
+def install_dependencies()
+  common = Common.new
+  common.docker.requires_docker
+
+  common.run_inline %W{docker-compose run --rm ui npm install}
+end
+
+def swagger_regen()
+  common = Common.new
+  common.docker.requires_docker
+
+  common.run_inline %W{docker-compose run --rm ui npm run codegen}
+end
+
+class DevUpOptions
+  ENV_CHOICES = %W{local test prod}
+  attr_accessor :env
+
+  def initialize
+    self.env = "test"
+  end
+
+  def parse args
+    parser = OptionParser.new do |parser|
+      parser.banner = "Usage: ./project.rb dev-up [options]"
+      parser.on(
+          "--environment ENV", ENV_CHOICES, "Environment [local (default), test, prod]") do |v|
```

Comment:
For what it's worth, I believe that providing multiple ways to specify an option is wrong for usability, despite how prevalent the practice. The `optparse` library will throw an error for `-e` but `--env` will work because it's close-enough to `--environment`.